OneginiExampleAppSwift/AppRouter.swift

```diff
@@ -52,7 +52,13 @@ protocol AppRouterProtocol: class {
 }
 
 class AppRouter: NSObject, AppRouterProtocol {
-    var window: UIWindow
+    var windowCreator: () -> UIWindow
+    private lazy var window: UIWindow = {
+        let window = windowCreator()
+        window.backgroundColor = UIColor.white
+        window.makeKeyAndVisible()
+        return window
+    }()
```

Comment:
Can we solve this with the lazy injection supported by Swinject? https://github.com/Swinject/Swinject/blob/master/Documentation/DelayedInjection.md